spec/functional/resource/dpkg_package_spec.rb

```diff
@@ -45,11 +45,17 @@ def set_dpkg_package_name(name)
     dpkg_package.package_name name
   end
 
-  def should_be_purged_or_removed(package, action)
+  def should_be_purged_or_removed(package, action=nil)
     if action == :purge
-      shell_out!("dpkg -s #{package}", returns: [1])
+        expect(shell_out(
+          'dpkg -s chef-integration-test').stdout).to match(/not-installed|no info/)
+    elsif action == :remove
+      expect(shell_out("dpkg -s #{package}").stdout).to match(/(deinstall ok config-files|not-installed)/)
```

Comment:
Down here this should have been left the way it was since i really wanted to assert that the package was in the deinstalled state, and not in the not-installed/exit(1) state.